File: 4074-count-subarrays-with-majority-element-i/count-subarrays-with-majority-element-i.cpp
class Solution {
public:
    int countMajoritySubarrays(vector<int>& nums, int target) {
        int sz = nums.size();
        int ans = 0;
        for (int i = 0; i < sz; ++i) {
            int cnt = 0;
            for (int j = i; j < sz; ++j) {
                int subArrayLength = j - i + 1;
                if (nums[j] == target) {
                    cnt++;
                }
                if (cnt > subArrayLength / 2) {
                    ans++;
                }
            }
        }
        return ans;
    }
};

File: 4074-count-subarrays-with-majority-element-i/count-subarrays-with-majority-element-i.py
class Solution:
    def countMajoritySubarrays(self, nums: List[int], target: int) -> int:
        
        n = len(nums)
        ans = 0
        for i in range(n):
            cnt = 0
            for j in range(i,n):
                subArrayLength = j-i+1
                if (nums[j]==target):
                    cnt+=1
                if (cnt>subArrayLength//2):
                    ans+=1
        

        return ans
        